enterprise/models.py

```diff
@@ -2097,14 +2097,32 @@ def is_audit_enrollment(self):
     @property
     def license(self):
         """
-        Returns the license associated with this enterprise course enrollment if one exists.
+        Returns the license fulfillment associated with this enterprise course enrollment if one exists.
         """
         try:
             associated_license = self.licensedenterprisecourseenrollment_enrollment_fulfillment  # pylint: disable=no-member
         except LicensedEnterpriseCourseEnrollment.DoesNotExist:
             associated_license = None
         return associated_license
 
+    @property
+    def learner_credit_fulfillment(self):
+        """
+        Returns the Learner Credit fulfillment associated with this enterprise course enrollment if one exists.
+        """
+        try:
+            associated_fulfillment = self.learnercreditenterprisecourseenrollment_enrollment_fulfillment  # pylint: disable=no-member
+        except LearnerCreditEnterpriseCourseEnrollment.DoesNotExist:
+            associated_fulfillment = None
+        return associated_fulfillment
+
+    @property
+    def fulfillment(self):
+        """
+        Find and return the related EnterpriseFulfillmentSource subclass, or None.
+        """
+        return self.license or self.learner_credit_fulfillment
```

Comment:
I do think adam might be right because the subclasses of EnterpriseFulfillmentSource become separate physical tables.  While the relationship between an ECE <-> LC fulfillment is 1:1, and ECE <-> Licensed fulfillment is also 1:1, that doesn't technically stop an ECE from being related to an LC and Licensed fulfillment simultaneously. I think we rely solely on business logic to protect against this case.